File: v2/src/main/java/com/api/v2/people/requests/PersonUpdatingDto.java
package com.api.v2.people.requests;

import com.api.v2.people.dtos.Address;
import com.api.v2.people.enums.Gender;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

public record PersonUpdatingDto(
        @NotBlank
        String firstName,
        String middleName,
        @NotBlank
        String lastName,
        @NotNull
        LocalDate birthDate,
        @NotBlank
        @Size(min = 10, max = 10)
        String phoneNumber,
        @NotNull
        Gender gender,
        @NotBlank
        @Email
        String email,
        @Valid
        Address address
) {
}

File: v2/src/main/kotlin/com/api/v2/people/requests/PersonUpdatingDTO.kt
package com.api.v2.people.requests

import com.api.v2.people.dtos.Address
import com.api.v2.people.enums.Gender
import jakarta.validation.Valid
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.time.LocalDate

data class PersonUpdatingDTO(
    @NotBlank
    val firstName: String,
    val middleName: String?,
    @NotBlank
    val lastName: String,
    @NotNull
    val birthDate: LocalDate,
    @NotBlank
    @Email
    val email: String,
    @NotBlank
    @Size(min = 10, max = 10)
    val phoneNumber: String,
    @NotNull
    val gender: Gender,
    @Valid
    val address: Address
)
